core/app/models/spree/promotion.rb

```diff
@@ -220,6 +220,20 @@ def used_by?(user, excluded_orders = [])
       end
     end
 
+    # Removes a promotion and any adjustments or other side effects from an
+    # order.
+    # @param order [Spree::Order] the order to remove the promotion from.
+    # @return [undefined]
```

Comment:
@gmacdougall thanks! Updated and opened a PR to remove more of these: https://github.com/solidusio/solidus/pull/1454